pam/pam.go

```diff
@@ -67,6 +79,61 @@ func sendReturnMessageToPam(mTx pam.ModuleTransaction, retStatus adapter.PamRetu
 	}
 }
 
+func initLogging(args map[string]string) (func(), error) {
+	level := log.InfoLevel
+	if args["debug"] == "true" {
+		level = log.DebugLevel
+	}
+	log.SetLevel(level)
+
+	if out, ok := args["logfile"]; ok && out != "" {
+		f, err := os.OpenFile(out, os.O_CREATE|os.O_APPEND|os.O_RDWR, 0600)
+		if err != nil {
+			return nil, err
+		}
+		log.SetOutput(f)
+		return func() { f.Close() }, nil
+	}
+
+	disableTerminalLogging := func() {
+		if log.IsLevelEnabled(log.DebugLevel) {
+			return
+		}
+		if term.IsTerminal(int(os.Stdin.Fd())) {
+			return
+		}
+		log.SetLevel(log.WarnLevel)
+	}
+
+	if !journal.Enabled() {
+		log.SetHandler(nil)
+		disableTerminalLogging()
+		return func() {}, nil
+	}
+	if args["disable_journal"] == "true" {
+		log.SetHandler(nil)
+		disableTerminalLogging()
+		return func() {}, nil
+	}
+
+	log.SetHandler(func(_ context.Context, level log.Level, format string, args ...interface{}) {
+		journalPriority := journal.PriNotice
+		switch level {
+		case log.DebugLevel:
+			journalPriority = journal.PriDebug
+		case log.InfoLevel:
+			journalPriority = journal.PriInfo
+		case log.WarnLevel:
+			journalPriority = journal.PriWarning
+		case log.ErrorLevel:
+			journalPriority = journal.PriErr
+		}
+		journal.Print(journalPriority, format, args...)
+	})
```

Comment:
Not sure if we want this for the daemon too (although it should be automatic in such case), but if we do maybe this could be part of `internal/log`